File: Брославський Юрій/Bill.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_1
{
    internal class Bill
    {
        private double LimitingAmount;
        private double CurrentDebt;
        private string CustomerName;

        public Bill(double limitingAmount)
        {
            LimitingAmount = limitingAmount;
            CurrentDebt = 0;
        }

        public bool Check(double amount)
        {
            return CurrentDebt + amount <= LimitingAmount;
        }

        public void Add(double amount)
        {
            if (Check(amount))
            {
                CurrentDebt += amount;
            }
            else
            {
                Console.WriteLine("Exceeded bill limit. Action canceled.");
            }
        }

        public void Pay(double amount, string customerName)
        {
            if (amount <= 0)
            {
                Console.WriteLine("Payment amount must be greater than zero.");
                return;
            }

            if (CurrentDebt == 0)
            {
                Console.WriteLine($"{customerName}'s bill is already paid.");
                return;
            }

            if (amount >= CurrentDebt)
            {
                CurrentDebt = 0;
                Console.WriteLine($"{customerName}'s bill fully paid.");
            }
            else
            {
                CurrentDebt -= amount;
                Console.WriteLine($"Paid {amount:C} towards {customerName}'s bill. Remaining debt: {CurrentDebt:C}");
            }
        }

        public void ChangeTheLimit(double amount)
        {
            LimitingAmount = amount;
            Console.WriteLine($"Bill limit changed to {amount:C}");
        }

        public double GetCurrentDebt()
        {
            return CurrentDebt;
        }
    }
}

File: Тисовський Олег/Patern-Lab-1/src/main/java/source/domain/Bill.java
package source.domain;

public class Bill {
    private double limitingAmount;
    private double currentDebt;

    public Bill(double limitingAmount, double currentDebt) {
        this.limitingAmount = limitingAmount;
        this.currentDebt = currentDebt;
    }

    public boolean check(double amount) {
        if(amount > limitingAmount)
            return false;
        else
            return true;
    }

    public void add(double amount) {
        this.currentDebt += amount;
    }

    public void pay(double amount) {
        this.currentDebt -= amount;
    }

    public void changeTheLimit(double newAmount) {
        this.limitingAmount = newAmount;
    }

    public double getLimitingAmount() {
        return limitingAmount;
    }

    public double getCurrentDebt() {
        return currentDebt;
    }

    public void setCurrentDebt(double currentDebt) {
        this.currentDebt = currentDebt;
    }

}
